lib/PhasarLLVM/TypeHierarchy/LLVMVFTable.cpp

```diff
@@ -48,4 +50,33 @@ nlohmann::json LLVMVFTable::getAsJson() const {
   return J;
 }
 
+std::vector<const llvm::Function *>
+LLVMVFTable::getVFVectorFromIRVTable(const llvm::ConstantStruct *VT) {
+  std::vector<const llvm::Function *> VFS;
+  for (const auto &Op : VT->operands()) {
+    if (auto *CA = llvm::dyn_cast<llvm::ConstantArray>(Op)) {
+      for (auto It = CA->operands().begin() + 2; It != CA->operands().end();
```

Comment:
```suggestion
      for (auto It = std::advance(CA->operands().begin(), 2); It != CA->operands().end();
```
+ This needs also a comment or constant name to make clear why we start not on the first but the third operand.

Reply:
What is the advantage of using std::advance? Bounds checking?
Index 0 is base offset, index 1 is RTTI, then the actual function entries start. But yeah, this should not just be a magic constant.

Reply:
No, there is no bounds checking (or other performance penalty). The advantage of `advance` is that it's handing the iterator correctly based on it's iterator categorie and does not solely rely on pointer arithmetic. It makes the code a bit more readable. Plus, it's easier to handle for some static analyzers :wink:, as some are not super good in handling pointer arithmetic.

For me, it was a magic constant :)

Reply:
Shouldn't it then be `std::next` ?. As far as I know, std::advance only increments in-place

Reply:
And the +2 is not correct? It is also not pointer arithmetic, but operator+, isn't it? I really don't see the added value to use these functions and not just + 2.

Reply:
The +2 only works for rancom-access iterators, not in general as these STL functions do

Reply:
@fabianbs96 Yes, I mixed them up in my head, it's `std::next` if you need the iterator returned.

@MMory No, `+2` is not always correct, everything depends on the iterator category. For example, where a RandomAccessIterator requires `operator+` to be implemented, Forward or Bidirectional iterators do only need to supply operator++ (and operator-- for the Bidirectional one). So, to not make mistakes or let the code break when underlying stuff get's changed/refactored underneath, using `std::next` or `std::advance` is advised.

Just to make this clear, this does not cost performance the std-library will pick the most optimal implementation for the specified iterator and the compiler will optimize everything, so in the end there should be no performance penalty.

Reply:
got it, changing it. Thanks, again something learned :)